partiql/tests/tuple_ops.rs

```diff
@@ -0,0 +1,128 @@
+use assert_matches::assert_matches;
+use partiql_ast_passes::error::AstTransformationError;
+use partiql_catalog::context::SystemContext;
+use partiql_catalog::{Catalog, PartiqlCatalog};
+use partiql_eval as eval;
+use partiql_eval::env::basic::MapBindings;
+use partiql_eval::error::{EvalErr, PlanErr};
+use partiql_eval::eval::{BasicContext, EvalPlan, EvalResult, Evaluated};
+use partiql_eval::plan::EvaluationMode;
+use partiql_logical as logical;
+use partiql_parser::{Parsed, ParserError, ParserResult};
+use partiql_value::{DateTime, Value};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+enum TestError<'a> {
+    #[error("Parse error: {0:?}")]
+    Parse(ParserError<'a>),
+    #[error("Lower error: {0:?}")]
+    Lower(AstTransformationError),
+    #[error("Plan error: {0:?}")]
+    Plan(PlanErr),
+    #[error("Evaluation error: {0:?}")]
+    Eval(EvalErr),
+}
+
+impl<'a> From<ParserError<'a>> for TestError<'a> {
```

Comment:
#[from] ParserError<'a>?

Reply:
We can't currently as  `ParserError`, `AstTransformationError`, `PlanErr`, and `EvalErr` don't impl `Error`.

Tracking that in #498